sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobOutputStream.java

```diff
@@ -277,7 +274,11 @@ protected void writeInternal(final byte[] data, int offset, int length) {
              */
             byte[] buffer = new byte[length];
             System.arraycopy(data, offset, buffer, 0, length);
-            sink.next(ByteBuffer.wrap(buffer));
+
+            IOException exception = this.sink.tryEmitNext(ByteBuffer.wrap(buffer));
+            if (exception != null) {
+                this.lastError = exception;
+            }
```

Comment:
how about try-catch block and just throw inside the try ? (they it shouldn't be called "try").

Reply:
yeah I will change the code to just throw the exception and make it try catch here

Reply:
resolved